python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py

```diff
@@ -0,0 +1,250 @@
+                opts_file.get(localPath=self.getSharedPath())
+            else:
+                raise ApplicationConfigurationError(None, "Opts file type %s not yet supported please contact Ganga devs if you require this support" % getName(opts_file))
+            self.post_prepare()
+
+        except Exception as err:
+            logger.debug("Err: %s" % str(err))
+            self.unprepare()
+            raise
+
+        return 1
+
+    def configure(self, masterappconfig):
+        """
+        Required even though nothing is done in this step for this App
+        """
+        # Lets test the inputs
+        opt_file = self.getOptsFile()
+        dir_name = self.getDir()
+        return (None, None)
+
+    def getOptsFile(self):
+        """
+        This function rweturns a sanitized absolute path to the self.myOpts file from user input
+        """
+        if self.myOpts:
+            if isinstance(self.myOpts, LocalFile):
+                ## FIXME LocalFile should return the basename and folder in 2 attibutes so we can piece it together, now it doesn't
+                full_path = expandfilename(path.join(self.myOpts.localDir, path.basename(self.myOpts.namePattern)), force=True)
+                if not path.exists(full_path):
+                    raise ApplicationConfigurationError(None, "Opts File: \'%s\' has been specified but does not exist please check and try again!")
+                self.myOpts = LocalFile(namePattern=path.basename(full_path), localDir=path.dirname(full_path))
+                return self.myOpts
+            elif isinstance(self.myOpts, DiracFile):
+                return self.myOpts
+            else:
+                raise ApplicationConfigurationError(None, "Opts file type %s not yet supported please contact Ganga devs if you require this support" % getName(self.myOpts))
+        else:
+            raise ApplicationConfigurationError(None, "No Opts File has been specified, please provide one!")
+
+    def getDir(self):
+        """
+        This function returns a sanitized absolute path of the self.directory method from user input
+        """
+        if self.directory:
+            return path.abspath(expandfilename(self.directory))
+        else:
+            raise ApplicationConfigurationError(None, "No Opts File has been specified, please provide one!")
+
+    def exec_cmd(self, cmd):
+        """
+        This method exectutes a command within the namespace of the project. The cmd is placed in a bash script which is executed within the env
+        Args:
+            cmd (str): This is the command(s) which are to be executed within the project environment and directory
+        """
+
+        cmd_file = tempfile.NamedTemporaryFile(suffix='.sh', delete=False)
+
+        cmd_wrapper = './run bash -c "%s"' % cmd
+
+        cmd_file.write(cmd_wrapper)
+        cmd_file.flush()
+
+        script_run = 'bash %s' % cmd_file.name
+
+        rc, stdout, stderr = _exec_cmd(script_run, self.getDir())
```

Comment:
Well at the very least you can get rid of the `bash` on the `script_run = ...` line and just set `executable` to `/bin/bash` in the call to `subprocess.Popen()`. Starting `bash` three times for every call isn't cheap.